lib/Conversion/CGGIToJaxite/CGGIToJaxite.cpp

```diff
@@ -0,0 +1,306 @@
+    rewriter.modifyOpInPlace(op, [&] {
+      op.setType(newFuncType);
+      // In addition to updating the type signature, we need to update the
+      // entry block's arguments to match the type signature
+      Block &block = op.getBody().getBlocks().front();
+      block.addArguments(serverKeyType, op.getLoc());
+      block.addArguments(paramsType, op.getLoc());
+    });
+    return success();
+  }
+};
+
+struct ConvertCGGIToJaxiteLut3Op : public OpConversionPattern<cggi::Lut3Op> {
+  ConvertCGGIToJaxiteLut3Op(mlir::MLIRContext *context)
+      : OpConversionPattern<cggi::Lut3Op>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      cggi::Lut3Op op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+    FailureOr<Value> result_server_key =
+        getContextualJaxiteServerKeySet(op.getOperation());
+    if (failed(result_server_key)) return result_server_key;
+    Value serverKey = result_server_key.value();
+
+    FailureOr<Value> result_params =
+        getContextualJaxiteParams(op.getOperation());
+    if (failed(result_params)) return result_params;
+    Value params = result_params.value();
+
+    int64_t truth_table_value = op.getLookupTableAttr().getUInt();
+    int8_t truth_table_value_int8 = static_cast<int8_t>(truth_table_value);
+    Value tt = b.create<arith::ConstantOp>(
+        op.getLoc(), b.getIntegerAttr(b.getI8Type(), truth_table_value_int8));
+
+    // The ciphertext parameters (a, b, c) are passed in reverse order from cggi
+    // to jaxite to mirror jaxite API
+    auto createLut3Op = rewriter.create<jaxite::Lut3Op>(
+        op.getLoc(), op.getOutput().getType(), op.getC(), op.getB(), op.getA(),
+        tt, serverKey, params);
+    rewriter.replaceOp(op, createLut3Op);
+    return success();
+  }
+};
+
+struct ConvertCGGIToJaxiteTrivialEncryptOp
+    : public OpConversionPattern<lwe::TrivialEncryptOp> {
+  ConvertCGGIToJaxiteTrivialEncryptOp(mlir::MLIRContext *context)
+      : OpConversionPattern<lwe::TrivialEncryptOp>(context, /*benefit=*/2) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      lwe::TrivialEncryptOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    FailureOr<Value> result = getContextualJaxiteParams(op.getOperation());
+    if (failed(result)) return result;
+    Value serverParams = result.value();
+    lwe::EncodeOp encodeOp = op.getInput().getDefiningOp<lwe::EncodeOp>();
+    if (!encodeOp) {
+      return op.emitError() << "Expected input to TrivialEncrypt to be the "
+                               "result of an EncodeOp, but it was "
+                            << op.getInput().getDefiningOp()->getName();
+    }
+    auto createConstantOp = rewriter.create<jaxite::ConstantOp>(
+        op.getLoc(), op.getOutput().getType(), encodeOp.getPlaintext(),
```

Comment:
Since the `lwe::EncodeOp` takes a `SignlessIntegerLike` as the plaintext value, should you also check that it is an i1 boolean value? (If you write a test case where it's not I imagine it should error out).

Reply:
Done PTAL lmk if any of the checks are superflous?

Reply:
that lgtm!